scripts/py_matter_idl/matter_idl/generators/kotlin/MatterClusters.jinja

```diff
@@ -0,0 +1,168 @@
+{%- macro encode_value(source, encodable, depth) -%}
```

Comment:
Not sure this is right place, considering it is in early stage, these API is incomplete, it is better to put explicit comment to describe it + experimental keyword on the top in case.

Reply:
See https://github.com/project-chip/connectedhomeip/blob/master/scripts/py_matter_idl/matter_idl/generators/java/ChipClusters_java.jinja.

The macros in the provided template are defined at the beginning because, in templating systems like Jinja2, it's a common practice to define macros before they are used, similar to defining functions before invoking them in many programming languages.